src/server/gc_worker.rs

```diff
@@ -316,6 +367,11 @@ impl<E: Engine> GcRunner<E> {
             "safe_point" => safe_point
         );
 
+        if !self.need_gc(ctx, safe_point) {
```

Comment:
If it needs to do gc, it will check need_gc again at L286?

Reply:
Yes, I didn't remove the old one. If you think it's necessary I'll remove it then.